Adding shared library on Unix

tdegeus opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

Fixes https://github.com/conda-forge/ghostscript-feedstock/issues/10

tdegeus committed:
Adding shared library on Unix

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

tdegeus commented:
@conda-forge-admin, please rerender

tdegeus commented:
Fixes https://github.com/conda-forge/ghostscript-feedstock/issues/10

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

tdegeus commented:
@leouieda @ocefpaf Working on two things here : the installation of the shared library as requested in https://github.com/conda-forge/ghostscript-feedstock/issues/10 and the permission issues in https://github.com/conda-forge/ghostscript-feedstock/issues/11 . Note that the permission issues are now reproduced even in the feedstock (don't know how to solve yet though)

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [27, 28]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

chrisburr commented on recipe/meta.yaml:
```diff
@@ -22,7 +24,19 @@ requirements:
+    - if [ ! -f ${PREFIX}/lib/libgs.so ]; then ln -s ${PREFIX}/lib/libgs.so.{{ major }} ${PREFIX}/lib/libgs.so; fi  # [linux]
+    - if [ ! -f ${PREFIX}/lib/libgs.dylib ]; then ln -s ${PREFIX}/lib/libgs.dylib.{{ major }} ${PREFIX}/lib/libgs.dylib; fi  # [osx]
     - gs --version
+    - test -f ${PREFIX}/lib/libgs.so.{{ major }}  # [linux]
```
You can use this variable:
```suggestion
    - test -f ${PREFIX}/lib/libgs${SHLIB_EXT}.{{ major }}  # [unix]
```

tdegeus replied:
Awesome, thanks!

tdegeus committed:
Fixing tests

tdegeus committed:
Fixing maxos

tdegeus committed:
Fixing typo

tdegeus commented:
Ready for review @leouieda @ocefpaf 

Note: ghostscript contains the 'mistake' that it name `libgs.dylib.9.53.3` whereas the convention on osx is `libgs.9.53.3.dylib`, but I don't think it is the feedstock that should address this issue.

tdegeus commented:
P.S. I really only modified `meta.yaml` and `build.sh`, the rest comes from conda. I just squashed at some point so that's why it might appeared as if I modified these files.

ocefpaf commented on recipe/build.sh:
```diff
@@ -3,5 +3,12 @@
 ./configure --prefix=${PREFIX}
 
 make -j${CPU_COUNT}
+make so -j${CPU_COUNT}
```
make it so ;-p, sorry, I could not resist.
